gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/resources/endpoint/mfa/MFAEnrollEndpoint.java

```diff
@@ -93,9 +92,20 @@ public MFAEnrollEndpoint(FactorManager factorManager,
     public void handle(RoutingContext routingContext) {
         HttpServerRequest req = routingContext.request();
         switch (req.method().name()) {
-            case "GET":
-                renderPage(routingContext);
-                break;
+            case "GET": {
+                if (routingContext.user() == null) {
+                    logger.warn("User must be authenticated to enroll MFA challenge.");
+                    routingContext.fail(401);
+                    return;
+                }
+                var context = new MfaFilterContext(routingContext, routingContext.get(ConstantKeys.CLIENT_CONTEXT_KEY), factorManager);
+                if (context.userHasMatchingFactors()) {
+                    redirectToAuthorize(routingContext);
+                } else {
+                    renderPage(routingContext);
+                }
+            }
```

Comment:
I would prefer to keep this block in the renderPage method (to follow the convention for other Endpoint).

Otherwise that seems good.